src/runtime/internal/dom.ts

```diff
@@ -90,10 +90,14 @@ export function attr(node: Element, attribute: string, value?: string) {
 }
 
 export function set_attributes(node: Element & ElementCSSInlineStyle, attributes: { [x: string]: string }) {
+	let a;
 	for (const key in attributes) {
 		if (key === 'style') {
 			node.style.cssText = attributes[key];
-		} else if (key in node) {
+		} else if (
+			//@ts-ignore
+			(a = Object.getOwnPropertyDescriptor(node.__proto__, key)) && a.set
```

Comment:
It _feels_ like it would be faster to just do `const descriptors = Object.getOwnPropertyDescriptors(node.__proto__)` once and then do `descriptors[key] && descriptors[key].set` here, but I don't actually have any data to back this up.

Reply:
Maybe you can't trust benchmarks but at least [this](https://esbench.com/bench/5da066734cd7e6009ef62825) supports your hunch, which makes sense to me. It certainly can't imagine it being any slower.